### PR TITLE
fixing #172

### LINKED ISSUE #172: Suspend an Instance
**Is your feature request related to a problem? Please describe.**
Be able to suspend, not stop, an instance. 

**Describe the solution you'd like**
Similar to gce_vm_suspend.

**Describe alternatives you've considered**
Adding a PR in a minute.

**Additional context**
Suspend is only supported in beta, so I'm not sure about how much you are supporting of that: https://cloud.google.com/compute/docs/reference/rest/beta/instances/suspend

### PR DESCRIPTION
**Pull request by muschellij2**
This should fix #172, but it relies on the beta version of the API.  I'm not sure the support or being able to track this as I know multiple API versions can be a hassle.

### CODE REVIEW

**Comment by muschellij2** [comment]
So I added tests, but I don't have a project up and running right not to try them out.  Also, there were a few lurking bugs and arguments that seem to be swirling around that I believe i fixed.

File: R/vms.R [hunk 1]
```diff
@@ -124,6 +127,23 @@ gce_vm <- function(name,
...
 }
 
+# using the partial list of rules at
+# https://cloud.google.com/compute/docs/labeling-resources 
+# and the conventions required in the GUI.
+validate_vm_name = function(name) {
+  ss = strsplit(name, "")[[1]]
+  assertthat::assert_that(
+    is.character(name),
+    length(name) == 1,
```
Comment (MarkEdmondson1234) [inline R/vms.R:137]: Could be assertthat::is.string() that covers length one characters.

**Comment by MarkEdmondson1234** [comment]
I'm going to merge this in to stop it getting so big, but have gone through and looks like it won't break anything :) - will set up some more tests before any publication to CRAN

**Comment by MarkEdmondson1234** [comment]
It did break something, but https://github.com/cloudyr/googleComputeEngineR/issues/181 fixed it.

**Pull request merged**